authnz/src/main/java/io/stargate/auth/AuthenticationSubject.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.auth;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface AuthenticationSubject {
+
+  String token();
+
+  String roleName();
+
+  boolean isFromExternalAuth();
+
+  static AuthenticationSubject of(String token, String roleName, boolean fromExternalAuth) {
+    return ImmutableAuthenticationSubject.builder()
+        .token(token)
+        .roleName(roleName)
+        .isFromExternalAuth(fromExternalAuth)
+        .build();
+  }
+
+  static AuthenticationSubject of(String token, String roleName) {
+    return ImmutableAuthenticationSubject.builder().token(token).roleName(roleName).build();
```

Comment:
Should we set `isFromExternalAuth` too?... It does not appear to have a default values, so I'd guess the `build()` call will error out :thinking:

Reply:
done